src/plugins/map/MapObject.h

```diff
@@ -206,34 +211,58 @@ class MapObjectMapLoc : public MapObject
 
 	virtual bool CanDisplayObject() const override { return true; }
 
+	bool				  m_isSelected = false; // exposed directly for imgui reference
+
 private:
 	void UpdateMapLoc();
 	void RemoveMapLoc();
-
 	void UpdateText();
 
-	bool                  m_isCreatedFromDefaultLoc;
+	bool                  m_isCreatedFromDefaultLoc = false;
 	bool                  m_initialized = false;
-	int                   m_index = 1;
+	int                   m_index = -1;
 	std::string           m_tag;
 	std::string           m_labelText;
-	float                 m_lineSize;
-	float                 m_width;
-	MQColor               m_color;
 	std::vector<MapViewLine*> m_lines;
-	float                 m_circleRadius;
-	MQColor               m_circleColor;
 	MapCircle             m_circle;
+	MapLocParams		  m_mapLocParams;
 };
 
-void UpdateDefaultMapLocParams();
-void UpdateDefaultMapLocInstances();
+class MapLocTemplate
+{
+public:
+	MapLocTemplate(const MapLocParams& params, const std::string& label,
+		const std::string& tag, const CVector3& pos, bool isDefault);
 
-void MakeMapLoc(const MapLocParams& params, const std::string& label,
-	const std::string& tag, const CVector3& pos, bool isDefault);
+	int GetIndex();
+	MapObjectMapLoc* GetMapLoc();
+	void MakeMapLocFromTemplate();
+	void ClearReferenceToMapLoc();
+	void UpdateFromParams(const MapLocParams& params);
+	bool IsCreatedFromDefaults() { return m_isCreatedFromDefaultLoc; }
+	void SetCreatedFromDefaults(bool isCreatedFromDefaults)
+	{
+		m_isCreatedFromDefaultLoc = isCreatedFromDefaults;
+		m_MapLoc->SetCreatedFromDefaults(m_isCreatedFromDefaultLoc);
+	}
+
+private:
+	MapLocParams		  m_mapLocParams;
```

Comment:
align variable names with spaces. I can see things are off because some are using tabs and they don't align the same.